ownCloud/Client/Viewer/PreviewViewController.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  PreviewViewController.swift
+//  ownCloud
+//
+//  Created by Michael Neuwert on 27.08.2019.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+* Copyright (C) 2018, ownCloud GmbH.
+*
+* This code is covered by the GNU Public License Version 3.
+*
+* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import UIKit
+import ownCloudSDK
+import QuickLook
+
+class PreviewViewController : DisplayViewController, QLPreviewControllerDataSource, QLPreviewControllerDelegate {
+
+	private var qlPreviewController: QLPreviewController?
+	var tapToHideBarsGestureRecognizer: UITapGestureRecognizer!
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		qlPreviewController = QLPreviewController()
+		addChild(qlPreviewController!)
+		qlPreviewController!.view.frame = self.view.bounds
+		self.view.addSubview(qlPreviewController!.view)
+		qlPreviewController!.didMove(toParent: self)
+
+		qlPreviewController?.view.isHidden = true
+	}
+
+	override func viewSafeAreaInsetsDidChange() {
+		super.viewSafeAreaInsetsDidChange()
+
+		if let qlPreviewController = self.qlPreviewController {
+			qlPreviewController.view.translatesAutoresizingMaskIntoConstraints = false
+
+			NSLayoutConstraint.activate([
+				qlPreviewController.view.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
+				qlPreviewController.view.bottomAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.bottomAnchor),
+				qlPreviewController.view.leadingAnchor.constraint(equalTo: self.view.leadingAnchor),
+				qlPreviewController.view.trailingAnchor.constraint(equalTo: self.view.trailingAnchor)
+				])
+		}
+
+		self.view.layoutIfNeeded()
+	}
+
+	override func renderSpecificView(completion: @escaping (Bool) -> Void) {
+		if source != nil {
+
+			self.qlPreviewController?.dataSource = self
+			self.qlPreviewController?.reloadData()
+
+			qlPreviewController?.view.isHidden = false
+
+			self.tapToHideBarsGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(self.tapToHideBars))
+			self.tapToHideBarsGestureRecognizer.delegate = self
+			self.tapToHideBarsGestureRecognizer.delaysTouchesBegan = true
+			self.qlPreviewController?.view.gestureRecognizers?.forEach({ $0.delegate = self })
+			self.qlPreviewController?.view?.addGestureRecognizer(self.tapToHideBarsGestureRecognizer)
+
+			completion(true)
+		} else {
+			completion(false)
+		}
+	}
+
+	@objc func tapToHideBars() {
```

Comment:
Likewise, since this method really both hides and shows the bars, maybe rename this to `showHideBars` (or similar) to make clearer what that method does.

Reply:
@felix-schwarz done